### PR TITLE
Pass basemap details to Vue components via env vars

### PR DESCRIPTION
**Pull request by m-mohr**
Implements #260

Is this going into the right direction, @soxofaan ?

~PS: This will probably only show an effect in a day or so as the new release of the Vue component library is still propagating through to the CDNs. It should work once https://www.jsdelivr.com/package/npm/@openeo/vue-components shows the 2.7.0 tag~

### CODE REVIEW

**Comment by m-mohr** [comment]
Works for me when I set env variables in Jupyter:

```
%env OPENEO_JUPYTER_BASEMAP_ATTRIBUTION=© <a href="https://www.mapbox.com/about/maps/">Mapbox</a> © <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> <strong><a href="https://www.mapbox.com/map-feedback/" target="_blank">Improve this map</a></strong>
%env OPENEO_JUPYTER_BASEMAP_URL=https://api.mapbox.com/styles/v1/mapbox/light-v10/tiles/{z}/{x}/{y}?access_token=...
```

**Comment by soxofaan** [comment]
I haven't tested this yet, but this the diff looks fine to me

two small questions:

- are both env vars independent from each other? E.g  it can make sense to set ATTRIBUTION without setting URL?
- I'm wondering if it's necessary to include `JUPYTER` in the env var name. I guess it's possible to use this feature outside of a jupyter notebook context too (e.g a python based web app that reuses the vue components)

**Comment by m-mohr** [comment]
> * are both env vars independent from each other? E.g  it can make sense to set ATTRIBUTION without setting URL?

Yes.

> * I'm wondering if it's necessary to include `JUPYTER` in the env var name.

If chosen the name because it is inside the jupyter.py (too) and `_repr_html_` is a Jupyter thing (is it?) - but happy to rename to something different. Any preferences?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Setting the environment variable `OPENEO_JUPYTER_BASEMAP_URL` allows to set a new templated URL to a XYZ basemap for the Vue Components library,  `OPENEO_JUPYTER_BASEMAP_ATTRIBUTION` allows to set the attribution for the basemap ([#260](https://github.com/Open-EO/openeo-python-client/issues/260))
```
Comment (m-mohr) [inline CHANGELOG.md:12]:
```suggestion
- Setting the environment variable `OPENEO_BASEMAP_URL` allows to set a new templated URL to a XYZ basemap for the Vue Components library,  `OPENEO_BASEMAP_ATTRIBUTION` allows to set the attribution for the basemap ([#260](https://github.com/Open-EO/openeo-python-client/issues/260))
```

File: openeo/internal/jupyter.py [hunk 2]
```diff
@@ -85,8 +86,18 @@ def render_component(component: str, data = None, parameters: dict = None):
     # Special handling for batch job results, show either item or collection depending on the data
     if component == "batch-job-result":
         component = "item" if data["type"] == "Feature" else "collection"
-    elif component == "data-table":
+    
+    if component == "data-table":
         parameters['columns'] = TABLE_COLUMNS[parameters['columns']]
+    elif (component in ['collection', 'collections', 'item', 'items']):
+        url = os.environ.get("OPENEO_JUPYTER_BASEMAP_URL")
```
Comment (m-mohr) [inline openeo/internal/jupyter.py:93]:
```suggestion
        url = os.environ.get("OPENEO_BASEMAP_URL")
```

File: openeo/internal/jupyter.py [hunk 3]
```diff
@@ -85,8 +86,18 @@ def render_component(component: str, data = None, parameters: dict = None):
...
+        attribution = os.environ.get("OPENEO_JUPYTER_BASEMAP_ATTRIBUTION")
```
Comment (m-mohr) [inline openeo/internal/jupyter.py:94]:
```suggestion
        attribution = os.environ.get("OPENEO_BASEMAP_ATTRIBUTION")
```

**Comment by soxofaan** [comment]
sidenote: I think it makes sense to cover this with a more general config system, to be discussed at #264  
(but PR #261 should not wait for that to be merged)

**Comment by soxofaan** [comment]
merged (manually) in 71090eb585279b36cfef68e6cced855a85ded87a

**Pull request closed without merging**